### DISCUSSION TITLE
Use correct table/figure references when child document (template) is rendered multiple times

### DISCUSSION THREAD

**Discussion by mattpollock** [quarto-dev/quarto-cli#10644] [Q&A]
I asked [a question on stackoverflow about this](https://stackoverflow.com/questions/78924971/how-to-get-correct-figure-references-when-a-quarto-child-document-is-rendered-mu) but it may be more of a "feature request" than a true SO question for how to do a thing. The following content is largely copied from the SO question:

When repeatedly using child documents with figures/tables all references are to the last version of the figure/table in the document/site. I set up a github repo to make [a minimum reproducible example](https://github.com/mattpollock/quarto-labels), since it's too much code to dump here. Just clone the repo and run `quarto render`.

-------

Details:

I have a quarto book project that uses the `knitr` engine. It conducts an analysis on several different days of data. To do this I have a quarto document for each day with parameters in the YAML header, e.g.,

```yaml
params:
  date: "2024-08-28"
```

In the body of the document I have 

    ```{r}
    options(knitr.duplicate.label = "allow")
    ```
    
    ```{r child='day-template.qmd'}
    ```

I generate a book (HTML or PDF) using `quarto render`. There is a chapter for each day, and the `day-template.qmd` analysis executed for each chapter. The problem is that even though I'm allowing duplicate labels, the figure references using those labels are not correct.

E.g., if there is a code chunk in `day-template.qmd` that generates a plot with the label `fig-a`. I can reference that figure in the body of `day-template.qmd` using `@fig-a`. However if `day-template.qmd` is creating content for chapters 2 and 3, then the `@fig-a` reference that is in chapter 2 will reference the version of the figure in chapter 3.

Here is what it looks like when rendered to PDF:

[![enter image description here][1]][1]

Is there a way to help quarto figure out how to properly configure figure labels in this case?

  [1]: https://i.sstatic.net/KlzYBdGy.png

**Comment by mattpollock** [comment]
My tentative conclusion is that this isn't yet supported. Is that correct? Are there plans to support this sort of pattern?

**Comment by iagopinal** [comment]
Hi, see if [figure-table-formatter](https://github.com/iagopinal/figure-table-formatter) can solve your issue. You can cross-reference the figures and tables in the order that they are mentioned in the text, and sort them accordingly. It automatically outputs a file with the sorted tags of the figures and tables. You can use this file to automatically sort the figures in other files using the same filter (handy for supplements).

It would be nice to have the same functionality in the native Quarto, but at least this is a viable alternative now.
